Instagram/View Controllers/ComposeViewController.m

```diff
@@ -12,35 +12,63 @@ @interface ComposeViewController () <UIImagePickerControllerDelegate, UINavigati
 
 @property (weak, nonatomic) IBOutlet UIImageView *imageView;
 @property (weak, nonatomic) IBOutlet UITextField *captionField;
+@property (weak, nonatomic) IBOutlet UIProgressView *progressView;
+@property (weak, nonatomic) IBOutlet UILabel *progressLabel;
 
 @end
 
 @implementation ComposeViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    [self hideProgress];
 }
 
 - (void)clear {
     UIImage *placeholderImage = [UIImage imageNamed:@"image_placeholder"];
     [self.imageView setImage:placeholderImage];
     [self.captionField setText:@""];
+    [self hideProgress];
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
 - (void)imagePickerController:(UIImagePickerController *)picker
 didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
     
     UIImage *editedImage = info[UIImagePickerControllerEditedImage];
-    [self.imageView setImage:editedImage];
+    UIImage *resizedImage = [self compress:editedImage];
+    [self.imageView setImage:resizedImage];
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
+- (UIImage *)compress:(UIImage *)image {
```

Comment:
This seems like an incredibly useful method that you'd probably want to reuse in the future. A good place to put it would be as a method on `UIImage *`. You can add methods to classes you don't own using an objc feature called a "Category". It would look like this

UIImage+Compression.h
```
@interface UIImage (Compression)
- (UIImage *)compressedImage;
@end
```

UIImage+Compression.m
```
@implementation UIImage (Compression)

- (UIImage) compressedImage {
    ...your logic here...
}

@end
```

At the call site it would look like

```
UIImage *compressedImage = [image compressedImage];
```